bin/newman.js

```diff
@@ -1,557 +1,136 @@
-            legacyMode = !_.includes(procArgv, 'run') &&
-        !_.includes(['--help', '-h', '--version', '-v', '-V'], procArgv[2]) &&
-        !_.isEmpty(slicedArgs),
-            reporterArgs,
-            rawArgs,
-            result,
-            checkForColor,
-            checkForVersion,
-            checkForCommand,
-            vPos,
-            validCommands = [];
+// Warn on invalid command and then exits.
+program.on('command:*', (command) => {
+    console.error(`\n  error: invalid command \`${command}\`\n`);
+    program.help();
+});
 
-        !legacyMode && !module.parent && (procArgv = slicedArgs);
-        rawArgs = separateReporterArgs(procArgv);
-        try {
-            if (!legacyMode) {
-                run(rawArgs.argv);
-                program.commands.forEach(function (command) {
-                    validCommands.push(command._name);
-                    if (command._name === 'run') {
-                        result = command;
-                    }
-                });
-            }
-            else {
-                legacy(rawArgs.argv);
-                result = program;
+/**
+ * Starts the script execution.
+ * callback is required when this is required as a module in tests.
+ *
+ * @param {String[]} argv - Argument vector.
+ * @param {?Function} callback - The callback function invoked on the completion of execution.
+ */
+function run (argv, callback) {
+    waterfall([
+        (next) => {
+            // cache original argv, required to parse nested options later.
+            program._originalArgs = argv;
+            // omit custom nested options, otherwise commander will throw unknown options error
+            next(null, util.omitNestedOptions(argv, '--reporter-'));
+        },
+        (args, next) => {
+            let error = null;
+
+            try {
+                program.parse(args);
             }
-
-            if (_.isEmpty(procArgv) || _.includes(procArgv, '-h') || _.includes(procArgv, '--help')) {
-                return result.outputHelp();
+            catch (err) {
+                error = err;
             }
-            checkForVersion = _.includes(rawArgs.argv, '--version') ||
-                _.includes(rawArgs.argv, '-v') ||
-                _.includes(rawArgs.argv, '-V');
-            checkForCommand = _.includes(validCommands, rawArgs.argv[0]);
+            next(error);
+        },
+        (next) => {
+            // throw error if no argument is provided.
+            const error = !program.args.length && new Error('no arguments provided');
 
-            !checkForVersion && !legacyMode && !checkForCommand &&
-            customError('\nNewman: Invalid command or parameter.\n\n' +
-            'Example:\nnewman run my-api.json -e variables.json\n\n' +
-            'For more information, run:\nnewman --help\n');
-        }
-        catch (error) {
-            return callback(_.set(error, 'help', program.outputHelp()));
+            next(error ? error : null);
```

Comment:
I know this shouldn’t be a code review PR, but still noting down a few suggestions...

Error here did not need a variable instantiation and could have been computed while passing next arghmenfs.